### PR TITLE
Feature/dashboard layout

### PR DESCRIPTION
**Pull request by miguel-merlin**
## Description
Added the Admin layout component.

image

### CODE REVIEW

**Comment by miguel-merlin** [comment]
Closes #119

File: src/components/layouts/AdminLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+};
+
+export default function AdminLayout({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const router = useRouter();
+
+  const menuItems: MenuItem[] = [
```
Comment (ulrokx) [inline src/components/layouts/AdminLayout.tsx:28]: Can we move this out of the function and into the module scope? If not this object will be created each render unnecessarily.

File: src/components/layouts/AdminLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    {
+      text: "Forms",
+      icon: <MailIcon />,
+      path: "/",
+    },
+  ];
+
+  const handleNavigation = async (item: MenuItem) => {
+    const path = item.path;
+    await router.push(path);
```
Comment (ulrokx) [inline src/components/layouts/AdminLayout.tsx:43]: Next router seems to return promises for some of these router methods but in all of their examples they don't await. I think this is fine here, although we might want to look at useTransition hook when this is hooked up to actual pages.

File: src/components/layouts/AdminLayout.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  };
+
+  return (
+    <div style={{ display: "flex" }}>
```
Comment (ulrokx) [inline src/components/layouts/AdminLayout.tsx:47]: There is nothing wrong with this but could we use Tailwind when possible? CSS in JS also has some performance downsides and by using Tailwind we avoid this potential slowdown and make the code consistent everywhere.

File: src/components/layouts/AdminLayout.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      <Drawer
+        variant="permanent"
+        style={{ width: DRAWER_WIDTH }}
+        sx={{
+          width: DRAWER_WIDTH,
+          flexShrink: 0,
+          "& .MuiDrawer-paper": {
```
Comment (ulrokx) [inline src/components/layouts/AdminLayout.tsx:54]: Can you do this with Tailwind as well? Example https://mui.com/base-ui/guides/working-with-tailwind-css/#adding-an-interactive-slider-component

**Review by ulrokx — CHANGES_REQUESTED** [review]
Nice

**Review by ulrokx — APPROVED** [review]
Great!

**Pull request merged**